code/game/gamemodes/game_mode.dm

```diff
@@ -201,9 +201,37 @@ var/global/list/additional_antag_types = list()
 		antag.update_current_antag_max(src)
 		antag.build_candidate_list(src) //compile a list of all eligible candidates
 
+	if(length(antag_templates) > 1) // If we have multiple templates to satisfy, we must pick candidates who satisfy fewer templates first, and fill the template with fewest candidates first
+		var/list/template_candidates = list()
+		var/list/all_candidates = list() // All candidates for every template, may contain duplicates
+		var/list/antag_templates_by_initial_spawn_req = list()
+
+		for(var/datum/antagonist/antag in antag_templates)
+			template_candidates[antag.id] = length(antag.candidates)
+			all_candidates += antag.candidates
```

Comment:
Should probably be `|=`, assuming that other antag templates can grab the same people as candidates.

Reply:
`all_candidates` needs to be able to contain duplicates, as it is used to generate `valid_templates_per_candidate`. If a candidate appears twice in `all_candidates`, then it means they are valid choices for two templates.

There's a comment on line 206 that mentions this, maybe the list could be named better? Nothing springs to mind.

Reply:
Ah my bad.